SMLHelper/Options/ConfigModOptions.cs

```diff
@@ -0,0 +1,700 @@
+        /// to generate a <see cref="ModOptionMetadata"/> and add it to the <see cref="modOptionsMetadata"/> dictionary.
+        /// </summary>
+        /// <param name="memberInfo">The <see cref="MemberInfo"/> of the member.</param>
+        /// <param name="memberType">The underlying <see cref="Type"/> of the member.</param>
+        private void ProcessFieldOrProperty(MemberInfo memberInfo, Type memberType)
+        {
+            if (memberType == typeof(bool))
+            {   // Generate a ModToggleOption
+                AddModOptionMetadata<ModToggleOption>(memberInfo, memberType);
+            }
+            else if (memberType == typeof(KeyCode))
+            {   // Generate a ModKeybindOption
+                AddModOptionMetadata<ModKeybindOption>(memberInfo, memberType);
+            }
+            else if (memberType.IsEnum || memberInfo.GetCustomAttributes(typeof(ChoiceAttribute), true).Any())
+            {   // Generate a ModChoiceOption
+                AddModOptionMetadata<ModChoiceOption>(memberInfo, memberType);
+            }
+            else if (new[] { typeof(float), typeof(double), typeof(int) }.Contains(memberType)
+                || memberInfo.GetCustomAttributes(typeof(SliderAttribute), true).Any())
+            {   // Generate a ModSliderOption
+                AddModOptionMetadata<ModSliderOption>(memberInfo, memberType);
+            }
+        }
+
+        /// <summary>
+        /// Processes the given method and hands off to <see cref="AddModOptionMetadata{TModOption}(MemberInfo, Type)"/>
+        /// to generate a <see cref="ModOptionMetadata"/> and add it to the <see cref="modOptionsMetadata"/> dictionary.
+        /// </summary>
+        /// <param name="methodInfo">The <see cref="MethodInfo"/> of the method.</param>
+        private void ProcessMethod(MethodInfo methodInfo)
+        {
+            if (methodInfo.GetCustomAttributes(true).Any(attribute => attribute is LabelAttribute || attribute is ButtonAttribute))
+            {   // Generate a ModButtonOption
+                AddModOptionMetadata<ModButtonOption>(methodInfo, null);
+            }
+        }
+
+        /// <summary>
+        /// Generates a <see cref="ModOptionMetadata"/> based on the member and its attributes, then adds it to the
+        /// <see cref="modOptionsMetadata"/> dictionary.
+        /// </summary>
+        /// <typeparam name="TModOption">The type of the <see cref="ModOption"/> to generate for this member.</typeparam>
+        /// <param name="memberInfo">The <see cref="MemberInfo"/> of the member.</param>
+        /// <param name="memberType">The underlying <see cref="Type"/> of the member.</param>
+        private void AddModOptionMetadata<TModOption>(MemberInfo memberInfo, Type memberType) where TModOption : ModOption
+        {
+            try
+            {
+                var modOptionType = typeof(TModOption); // cache the ModOption type for comparisons
```

Comment:
You don't need to cache this.  
You are using a Generic, so the type is handled at compile time.  
`typeof` is not a method, it's a keyword. Using it doesn't incur a a runtime cost.

Reply:
Good to know, thank you

Reply:
https://github.com/SubnauticaModding/SMLHelper/pull/173/commits/88fbdffba0748285567429b4fdb7d1981ad5d731